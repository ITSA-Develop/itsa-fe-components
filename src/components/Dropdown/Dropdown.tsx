import { Dropdown as AntDropdown, DropdownProps } from 'antd';

export const Dropdown = ({ ...rest }: DropdownProps) => {
	return <AntDropdown {...rest} />;
};
