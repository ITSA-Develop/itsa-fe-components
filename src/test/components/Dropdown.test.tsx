import { fireEvent, render, screen } from '@testing-library/react';
import { Dropdown } from '../../components/Dropdown/Dropdown';

describe('Dropdown component', () => {
	const menuItems = [
		{
			key: '1',
			label: 'First Item',
		},
		{
			key: '2',
			label: 'Second Item',
		},
	];

	it('renders without crashing', () => {
		const { container } = render(
			<Dropdown menu={{ items: menuItems }}>
				<button>Open Menu</button>
			</Dropdown>,
		);
		expect(screen.getByText('Open Menu')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('shows menu items when clicked', () => {
		const { container } = render(
			<Dropdown menu={{ items: menuItems }} trigger={['click']}>
				<button>Open Menu</button>
			</Dropdown>,
		);

		const button = screen.getByText('Open Menu');
		fireEvent.click(button);

		expect(screen.getByText('First Item')).toBeInTheDocument();
		expect(screen.getByText('Second Item')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
