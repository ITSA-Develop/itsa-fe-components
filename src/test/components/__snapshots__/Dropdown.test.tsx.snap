// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dropdown component > renders without crashing 1`] = `
<div>
  <button
    class="ant-dropdown-trigger"
  >
    Open Menu
  </button>
</div>
`;

exports[`Dropdown component > shows menu items when clicked 1`] = `
<div>
  <button
    class="ant-dropdown-trigger ant-dropdown-open"
  >
    Open Menu
  </button>
</div>
`;
